WriterDesk1/src/pages/SignUp.js

```diff
@@ -203,6 +211,16 @@ const SignUp = () => {
                                 buttonAgree={<Button onClick={() => {setShowUserDataPopup(false)}}>I understand</Button>}
                             />}
                         </div>
+                        <div style={{display: 'flex', alignSelf: 'flex-end', verticalAlign: 'middle'}}>
+                            <Checkbox sx={{alignSelf: 'center'}} onChange={(e) => {setAcceptNeceData(e.target.checked)}} />
+                            <Typography sx={{alignSelf: 'center', alignContent:'inline'}}>
+                            I allow the storage of <a className='userDataLinkPopup' onClick={() => {setShowNeceDataPopup(true)}} >necessary data</a>. and the execution of the application on this necessary data.
```

Comment:
Zou naar deze string kijken, want vindt hem alsno gniet geweldig

Reply:
"I allow the storage and execution of necessary application data" is denk ik beter